impl/src/main/java/org/jboss/forge/roaster/model/impl/JavaInterfaceMethodImpl.java

```diff
@@ -0,0 +1,184 @@
+package org.jboss.forge.roaster.model.impl;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+
+import org.eclipse.jdt.core.dom.Modifier.ModifierKeyword;
+import org.jboss.forge.roaster.model.Visibility;
+import org.jboss.forge.roaster.model.source.JavaInterfaceSource;
+import org.jboss.forge.roaster.model.source.MethodSource;
+import org.jboss.forge.roaster.model.util.Methods;
+
+public class JavaInterfaceMethodImpl extends MethodImpl<JavaInterfaceSource>
+{
+
+   public JavaInterfaceMethodImpl(JavaInterfaceSource parent)
+   {
+      super(parent);
+   }
+
+   public JavaInterfaceMethodImpl(JavaInterfaceSource parent, Method reflectMethod)
+   {
+      this(parent);
+      // Set method return type
+      if (reflectMethod.getReturnType() == Void.TYPE)
+      {
+         setReturnTypeVoid();
+      }
+      else
+      {
+         setReturnType(reflectMethod.getReturnType());
+      }
+      // Set method name
+      setName(reflectMethod.getName());
+      // Set method parameters
+      Class<?>[] paramTypes = reflectMethod.getParameterTypes();
+      String[] paramNames = Methods.generateParameterNames(paramTypes);
+      for (int i = 0; i < paramTypes.length; i++)
+      {
+         addParameter(paramTypes[i], paramNames[i]);
+      }
+   }
+
+   public JavaInterfaceMethodImpl(JavaInterfaceSource parent, Object internal)
+   {
+      super(parent, internal);
+   }
+
+   public JavaInterfaceMethodImpl(JavaInterfaceSource parent, String method)
+   {
+      super(parent, method);
+   }
+
+   @Override
+   public boolean isAbstract()
+   {
+      return true;
+   }
+
+   @Override
+   public MethodSource<JavaInterfaceSource> setAbstract(boolean abstrct)
+   {
+      if (abstrct)
+      {
+         modifiers.addModifier(method, ModifierKeyword.ABSTRACT_KEYWORD);
+         return this;
+      }
+      else
+      {
+         throw new IllegalStateException("Method interface should be 'abstract'");
+      }
+   }
+
+   @Override
+   public boolean isPublic()
+   {
+      return true;
+   }
+
+   @Override
+   public MethodSource<JavaInterfaceSource> setBody(String body)
```

Comment:
Not sure we should change this method. What happens if you call setStatic or setDefault after setBody?